### ISSUE TITLE
Control flow breaks inside event handler lambda functions - intended or not?

### ISSUE THREAD

**Issue by dummdidumm** [sveltejs/language-tools#876]
Control flow for TypeScript now should work correctly almost everywhere. One exception are event handlers when they use lambda functions.

**To Reproduce**
```svelte
<script lang="ts">
  let value: string | null = null;

  function acceptsString(value: string) {
    console.log(value);
  }
</script>

{#if value}
  <!-- Argument of type 'string | null' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'. -->
  <button on:click={() => acceptsString(value)} />
{/if}
```

**Expected behavior**
Now the question is: Is this a bug or standard TypeScript behavior?

One could argue it's a bug because one never gets to the click handler if the value is `null`.

One could also argue it's standard TypeScript behavior because the following React snippet will also yell at you:

```tsx
function Component() {
    let value: string | null = null;

    function acceptsString(value: string) {
        console.log(value);
    }

    return value && <p onClick={() => acceptsString(value)}>
        Start editing to see some magic happen :)
    </p>;
}
```

The reason is that TypeScript does not know when that lambda-function is invoked, so it cannot say for sure that `value` has not changed in the meantime because it's not a `const`.

Thoughts on this welcome!

**System (please complete the following information):**
Svelte for VS Code 104.6.4 / `svelte-check` svelte-check-1.2.5

**Comment by pushkine** [comment]
The svelte syntax mimics if statements rather than conditionals
So if `if (value)` in Typescript narrows the type correctly, `{#if value}` should too

**Comment by Buraburaite** [comment]
> One could argue it's a bug because one never gets to the click handler if the value is null.

I would agree. I am making a component that morphs based on the type of the provided variable, and I really expected that this would work.

Not my preference, but allowing for some kind of `// @ts-ignore` or `as any` would have been sufficient in my use case.

**Comment by dummdidumm** [comment]
#1657 was opened with the same underlying reason, seeing it as a bug
